### PR TITLE
Adding tests

### PR DESCRIPTION
**Pull request by soldni**

### CODE REVIEW

File: papermage/magelib/box.py [hunk 1]
```diff
@@ -73,6 +76,17 @@ def from_xy_coordinates(
     def xy_coordinates(self) -> Tuple[float, float, float, float]:
         return self.l, self.t, self.l + self.w, self.t + self.h
 
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Box):
+            return False
```
Comment (kyleclo) [inline papermage/magelib/box.py:81]: if comparing not the same type, can it error out instead?

File: papermage/magelib/entity.py [hunk 2]
```diff
@@ -46,10 +46,11 @@ def to_json(self) -> Dict:
 
     @classmethod
     def from_json(cls, entity_json: Dict) -> "Entity":
+        # the .get(..., None) or [] pattern is to handle the case where the key is present but the value is None
         return cls(
-            spans=[Span.from_json(span_json=span_json) for span_json in entity_json.get("spans", [])],
-            boxes=[Box.from_json(box_json=box_json) for box_json in entity_json.get("boxes", [])],
-            metadata=Metadata.from_json(entity_json.get("metadata", {})),
+            spans=[Span.from_json(span_json=span_json) for span_json in entity_json.get("spans", None) or []],
```
Comment (kyleclo) [inline papermage/magelib/entity.py:51]: can we avoid the `None` behavior? i think it's broken ppl's iterations in the past

Comment (soldni) [reply papermage/magelib/entity.py:51]: so it should just fail if metadata is set to none?

**Review by kyleclo — APPROVED** [review]
lgtm; pls check out the 2 comments

File: papermage/magelib/box.py (hunk 1 above)
Comment (soldni) [reply papermage/magelib/box.py:81]: returning false is default behavior in Python! comparisons usually don't raise errors.

**Pull request merged**